tests/integration/test_integration.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from pathlib import Path
+
+import pytest
+import requests
+import yaml
+from alertmanager import Alertmanager
+from pytest_operator.plugin import OpsTest  # type: ignore[import]  # noqa: F401
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+ALERTMANAGER_CONFIGURER_APP_NAME = METADATA["name"]
+ALERTMANAGER_APP_NAME = "alertmanager-k8s"
+WAIT_FOR_STATUS_TIMEOUT = 1 * 60
+DUMMY_HTTP_SERVER_PORT = 80
+
+
+class TestAlertmanagerConfigurerOperatorCharm:
+    @pytest.fixture(scope="module")
+    @pytest.mark.abort_on_fail
+    async def setup(self, ops_test: OpsTest):
+        await ops_test.model.set_config({"update-status-hook-interval": "2s"})
+        await self._deploy_alertmanager_k8s(ops_test)
+        charm = await ops_test.build_charm(".")
+        resources = {
+            f"{ALERTMANAGER_CONFIGURER_APP_NAME}-image": METADATA["resources"][
+                f"{ALERTMANAGER_CONFIGURER_APP_NAME}-image"
+            ]["upstream-source"],
+            "dummy-http-server-image": METADATA["resources"]["dummy-http-server-image"][
+                "upstream-source"
+            ],
+        }
+        await ops_test.model.deploy(
+            charm,
+            resources=resources,
+            application_name=ALERTMANAGER_CONFIGURER_APP_NAME,
+            trust=True,
+        )
+
+    @pytest.mark.abort_on_fail
+    async def test_given_alertmanager_configurer_charm_is_not_related_to_alertmanager_when_charm_deployed_then_charm_goes_to_blocked_status(  # noqa: E501
+        self, ops_test: OpsTest, setup
+    ):
+        await ops_test.model.wait_for_idle(
+            apps=[ALERTMANAGER_CONFIGURER_APP_NAME],
+            status="blocked",
+            timeout=WAIT_FOR_STATUS_TIMEOUT,
+        )
+
+    # @pytest.mark.abort_on_fail
+    # async def test_given_alertmanager_configurer_charm_in_blocked_status_when_alertmanager_relation_created_then_charm_goes_to_active_status(  # noqa: E501, W505
+    #     self, ops_test: OpsTest, setup
+    # ):
+    #     await ops_test.model.add_relation(
+    #         relation1=f"{ALERTMANAGER_CONFIGURER_APP_NAME}",
+    #         relation2=f"{ALERTMANAGER_APP_NAME}:remote-configurer",
+    #     )
+    #     await ops_test.model.wait_for_idle(
+    #         apps=[ALERTMANAGER_CONFIGURER_APP_NAME],
+    #         status="active",
+    #         timeout=WAIT_FOR_STATUS_TIMEOUT,
+    #     )
```

Comment:
Do we intend to use this test or is it a leftover?

Reply:
We are. As soon as `alertmanager-k8s` actually implements the required relation.